main.py

```diff
@@ -0,0 +1,319 @@
+                dew_point = float(data[4]) if data[4] else None
+                mean_dew_point = float(data[5]) if data[5] else None
+                min_dew_point = float(data[6]) if data[6] else None
+                max_humidity = int(data[7]) if data[7] else None
+                mean_humidity = int(data[8]) if data[8] else None
+                min_humidity = int(data[9]) if data[9] else None
+                max_pressure = float(data[10]) if data[10] else None
+                mean_pressure = float(data[11]) if data[11] else None
+                min_pressure = float(data[12]) if data[12] else None
+                max_visibility = float(data[13]) if data[13] else None
+                mean_visibility = float(data[14]) if data[14] else None
+                min_visibility = float(data[15]) if data[15] else None
+                max_wind_speed = int(data[16]) if data[16] else None
+                mean_wind_speed = int(data[17]) if data[17] else None
+                max_gust_speed = int(data[18]) if data[18] else None
+                precipitation = float(data[19]) if data[19] else None
+                cloud_cover = int(data[20]) if data[20] else None
+                events = data[21]
+                wind_direction_degrees = int(data[22]) if data[22] else None
+
+                return WeatherReading(date, max_temp, mean_temp, min_temp,
+                                      dew_point, mean_dew_point, min_dew_point,
+                                      max_humidity, mean_humidity,
+                                      min_humidity, max_pressure,
+                                      mean_pressure, min_pressure,
+                                      max_visibility, mean_visibility,
+                                      min_visibility, max_wind_speed,
+                                      mean_wind_speed, max_gust_speed,
+                                      precipitation, cloud_cover, events,
+                                      wind_direction_degrees)
+            except ValueError:
+                # Error occurred while converting data types, skip this line
+                return None
+        else:
+            # Line doesn't contain the expected number of fields, skip it
+            return None
+
+
+class WeatherResultsCalculator:
+    @staticmethod
+    def calculate_yearly_extremes(readings, year):
+        highest_temp = None
+        highest_temp_day = None
+        lowest_temp = None
+        lowest_temp_day = None
+        highest_humidity = None
+        highest_humidity_day = None
+
+        for reading in readings:
+            # Get year from the reading date (date format: YYYY-MM-DD)
+            reading_year = reading.date.split('-')[0]
+            if reading_year == year:
+                if reading.max_temp is not None and (highest_temp is None or
+                                                     reading.max_temp >
+                                                     highest_temp):
```

Comment:
```suggestion
                if reading.max_temp and reading.max_temp > highest_temp:
```
reading.max_temp will be considered true it has a value. 
Setting the highest_temp value to - infinity or some big negative value is better than None
Make this change and fix all checks